src/main/java/duke/command/TaskManager.java

```diff
@@ -0,0 +1,248 @@
+            if (!(index - 1 >= tasks.size()) && !(tasks.get(index - 1).getStatusIcon().equals(STATUS_DONE))) {
+                validIndexes[count] = index;
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return null;
+        }
+        return Arrays.copyOf(validIndexes, count);
+    }
+
+    public int[] filterValidDeleteIndexes(int[] indexes) {
+        int[] validIndexes = new int[indexes.length];
+        int count = 0;
+
+        for (int index : indexes) {
+            if (!(index - 1>= tasks.size()) ) {
+                validIndexes[count] = index;
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return null;
+        }
+        return Arrays.copyOf(validIndexes, count);
+    }
+
+    public int[] filterDoneIndexes(int[] indexes) {
+        int[] doneIndexes = new int[indexes.length];
+        int count = 0;
+
+        for (int index : indexes) {
+            if (!(index - 1 >= tasks.size()) && tasks.get(index - 1).getStatusIcon().equals(STATUS_DONE)) {
+                doneIndexes[count] = index;
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return null;
+        }
+        return Arrays.copyOf(doneIndexes, count);
+    }
+
+    public void setAsDone(String taskInfo) {
+        int[] indexes = filterIndexes(taskInfo);
+        int[] outOfRangeIndexes = filterOutOfRangeIndexes(indexes);
+        int[] validIndexes = filterValidIndexes(indexes);
+        int[] doneIndexes = filterDoneIndexes(indexes);
+
+        if (validIndexes != null) {
+            for (int validIndex : validIndexes) {
+                tasks.get(validIndex - 1).markAsDone();
+            }
+        }
+
+        try {
+            FileManager.updateFile(tasks);
+        } catch (IOException e) {
+            DisplayManager.printErrorMarkDoneUpdate();
+        }
+        DisplayManager.printSetAsDoneResult(tasks, outOfRangeIndexes, validIndexes, doneIndexes);
+    }
+
+    public void deleteTask(String taskInfo) {
+        int[] indexes = filterIndexes(taskInfo);
+        int[] outOfRangeIndexes = filterOutOfRangeIndexes(indexes);
+        int[] validIndexes = filterValidDeleteIndexes(indexes);
+        ArrayList<Task> deletedTasks = new ArrayList<>();
+
+        if (validIndexes != null) {
+            for (int validIndex : validIndexes) {
+                deletedTasks.add(tasks.get(validIndex - 1));
+                tasks.set(validIndex - 1, null);
+            }
+            tasks.removeIf(Objects::isNull);
+        }
+
+        try {
+            FileManager.updateFile(tasks);
+        } catch (IOException e) {
+            DisplayManager.printErrorDeleteUpdate();
+        }
+        DisplayManager.printDeleteTasksResult(deletedTasks, outOfRangeIndexes, tasks.size());
+    }
+
+    public void getAndPrintTaskList() {
+        if (tasks.size() == 0) {
+            DisplayManager.printErrorList();
```

Comment:
0 items in a taskList technically is a valid behaviour of the program. Should the naming of this method be different?